src/main/java/io/vertx/rabbitmq/impl/RabbitMQClientImpl.java

```diff
@@ -400,6 +404,9 @@ public void stop(Handler<AsyncResult<Void>> resultHandler) {
         log.debug("channel is close, try create Channel");
 
         channel = connection.createChannel();
+
+        if(channelConfirms)
+          channel.confirmSelect();
```

Comment:
Actually it's needed for reconnect because you recreate the channel. You can call confirm select once at startup. And then use waitForConfirms per request. Huge performance boost.